src/console/pulseq_interpreter/sequence_provider.py

```diff
@@ -363,7 +370,7 @@ def add_adc_gate(self, block: SimpleNamespace, gate: np.ndarray, clk_ref: np.nda
         """
         delay = max(int(block.delay * self.spcm_freq), int(block.dead_time * self.spcm_freq))
         adc_dur = block.num_samples * block.dwell
-        adc_len = int(adc_dur * self.spcm_freq)
+        adc_len = round(adc_dur * self.spcm_freq)
```

Comment:
Would it create mismatch between actual adc length and the adjusted value, similar to what we have observed for the adc length in rx_device? Is it something that we can tolerate or we can implement similar approach to rx_device?

Reply:
The problem here was that we were missing a single sample point for some ADC durations. This is because `int()` truncates the floating point value instead of rounding it. The calculated number of gate_samples was adjusted accordingly (see below).

Reply:
Does this clarify your concern? (Also the one from below?)